main.py

```diff
@@ -127,19 +189,35 @@ def query_callback(update, context):
             raise ValueError(f'Invalid query!\n{query}')
 
 
+        query.answer(text='Settings saved!')
         if data in [c.name for c in dixit_game.end_criteria]:
             dixit_game.end_criterion = dixit_game.end_criteria[data]
 
         query.edit_message_text(text=text, reply_markup=markup)
 
-    if query.data.isdecimal(): # if the user is sending us endgame values
-        dixit_game.end_criterion_number = int(query.data)
+    if query.data.startswith('end number'):
```

Comment:
Now that every type of query data is of the form `type:value` we could split it in `type` and `values` early, to avoid repeating code

Reply:
That's a good ideia!